### PR TITLE
MIRI review changes

### PR DESCRIPTION
**Pull request by jemorrison**
Summary of changes
1.  For MIRI data: adjusted pedestal to be pedestal = group2 - pedestal.  Kept the original pedestal for marking saturated pixels
2. added 2 plotting routines for noisy pixels
3. added a routine: read_slope_integration - to replace read_slope_file to handle reading in multiple integrations when there are different number of integration in different exposures. Vstack did not work for multiple integration data when the number of integrations varied between exposures 
4. Pulled out saturated data based on DQ flag of group 1 from the low_pedestal flagging

### CODE REVIEW

File: jwst_reffiles/dark_current/badpix_from_darks.py [hunk 1]
```diff
@@ -826,75 +820,115 @@ def read_slope_files(filenames):
...
+         output file of the plot
+
+    Returns
+    -------
+      prints the plot to disk
+
+    """
+    fig = plt.figure(figsize=(9, 9))
+    ax1 = fig.add_subplot(1, 1, 1)
+    im = ax1.imshow(image, extent=[0, 1023, 0, 1023], interpolation='None',
```
[the remark this answers is not in the document]
Comment (jemorrison) [reply jwst_reffiles/dark_current/badpix_from_darks.py]: fixed this to be the size of image

File: jwst_reffiles/dark_current/badpix_from_darks.py [hunk 2]
```diff
@@ -826,75 +820,115 @@ def read_slope_files(filenames):
...
+    std = np.std(data_array)
     xhigh = ave + std*sigma_threshold
 
     # plot histogram
 
-    fig = plt.figure(figsize=(9,9))
-    ax1 = fig.add_subplot(1,1,1)
+    fig = plt.figure(figsize=(9, 9))
+    ax1 = fig.add_subplot(1, 1, 1)
     ax1.hist(data_array, bins=nbins)
```
[the remark this answers is not in the document]
Comment (jemorrison) [reply jwst_reffiles/dark_current/badpix_from_darks.py]: a log scale was added as an option for histogram plots

File: jwst_reffiles/dark_current/badpix_from_darks.py [hunk 3]
```diff
@@ -160,62 +186,146 @@ def find_bad_pix(filenames, clipping_sigma=5., max_clipping_iters=5, noisy_thres
...
+    # noisy_new = np.logical_or((clean_std_slope > cut_limit),
+    #                          np.isnan(clean_std_slope))
+    noisy_new = clean_std_slope > cut_limit
+    num_noisy2 = len(np.where(noisy_new)[0])
+
+    if plot:
+        # plot the number of good slopes per pixel
+        max_values = np.amax(num_good)
+        plot_image(num_good, max_values, "Number of Good slopes/pixel ",
+                   "clean_pixel_number..png")
```
Comment (bhilbert4) [inline jwst_reffiles/dark_current/badpix_from_darks.py]: Double .. in output filename

**Comment by jemorrison** [comment]
good suggestion on the outdir for the plots. I added that.

**Pull request merged**